### PR TITLE
Changes to all the constructors to remove the args argument

### PR DESCRIPTION
**Pull request by LeonY1**
Removing args from the constructor
---
Part-of: #9942 

The apply_default decorator forces all the arguments to be keyword arguments; however, there are currently instances where there is the argument `*args` being used within the code. This PR removes all instances of using `*args`.

Let me know if I have to add anything to the CHANGELOG.txt or anything of that sort.

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by LeonY1** [comment]
@mik-laj Are you the one who is supposed to review this PR?

File: airflow/providers/amazon/aws/sensors/glue.py [hunk 1]
```diff
@@ -39,9 +39,8 @@ def __init__(self,
                  job_name,
                  run_id,
                  aws_conn_id='aws_default',
-                 *args,
                  **kwargs):
-        super(AwsGlueJobSensor, self).__init__(*args, **kwargs)
+        super(AwsGlueJobSensor, self).__init__(**kwargs)
```
Comment (mik-laj) [inline airflow/providers/amazon/aws/sensors/glue.py]:
```suggestion
        super().__init__(**kwargs)
```

**Comment by mik-laj** [comment]
How do you feel about marking all arguments as positional?

**Comment by LeonY1** [comment]
How would I go about marking them as positional? Honestly, from all the Python that I've done, I think keyword arguments feels a lot nicer than positional arguments. It's a lot more verbose in terms of showing what each argument is.

**Comment by mik-laj** [comment]
I meant the keyword argument. I think we should add an asterisk in the constructor. 
See: https://www.python.org/dev/peps/pep-3102/

```
def test_func(*, key=None):
    ...
```
This function only accepts keyword arguments.

**Comment by LeonY1** [comment]
Alright that can work. How do I know which keyword args are valid?

**Comment by LeonY1** [comment]
The `BaseOperator` has many different arguments if we're trying to work with those. However, I can see that this would make more sense to customize based off the current operator since they don't necessarily match every single operator.

```python
@apply_defaults
    def __init__(
        self,
        task_id: str,
        owner: str = conf.get('operators', 'DEFAULT_OWNER'),
        email: Optional[Union[str, Iterable[str]]] = None,
        email_on_retry: bool = conf.getboolean('email', 'default_email_on_retry', fallback=True),
        email_on_failure: bool = conf.getboolean('email', 'default_email_on_failure', fallback=True),
        retries: Optional[int] = conf.getint('core', 'default_task_retries', fallback=0),
        retry_delay: timedelta = timedelta(seconds=300),
        retry_exponential_backoff: bool = False,
        max_retry_delay: Optional[datetime] = None,
        start_date: Optional[datetime] = None,
        end_date: Optional[datetime] = None,
        depends_on_past: bool = False,
        wait_for_downstream: bool = False,
        dag=None,
        params: Optional[Dict] = None,
        default_args: Optional[Dict] = None,  # pylint: disable=unused-argument
        priority_weight: int = 1,
        weight_rule: str = WeightRule.DOWNSTREAM,
        queue: str = conf.get('celery', 'default_queue'),
        pool: Optional[str] = None,
        pool_slots: int = 1,
        sla: Optional[timedelta] = None,
        execution_timeout: Optional[timedelta] = None,
        on_execute_callback: Optional[Callable] = None,
        on_failure_callback: Optional[Callable] = None,
        on_success_callback: Optional[Callable] = None,
        on_retry_callback: Optional[Callable] = None,
        trigger_rule: str = TriggerRule.ALL_SUCCESS,
        resources: Optional[Dict] = None,
        run_as_user: Optional[str] = None,
        task_concurrency: Optional[int] = None,
        executor_config: Optional[Dict] = None,
        do_xcom_push: bool = True,
        inlets: Optional[Any] = None,
        outlets: Optional[Any] = None,
        **kwargs
    ):
```

**Comment by mik-laj** [comment]
We don't have to choose specific arguments. It is enough for us to make the following change in each operator.
```diff
- def __init__(self, table: str, keys: Dict[str, str], cassandra_conn_id: str, **kwargs: Any) -> None:
+ def __init__(self, *, table: str, keys: Dict[str, str], cassandra_conn_id: str, **kwargs: Any) -> None:
```

However, we can do it in a separate PR.

Now I would like to ask you to fix the pylint problems. We cannot accept this change without it.

**Comment by LeonY1** [comment]
@mik-laj I fixed the pylint issue

**Comment by LeonY1** [comment]
To fix the qubole error, would it be best to directly use the non-deprecated ValueCheckOperator?

**Comment by mik-laj** [comment]
*args cannot follow keyword arguments. This code is invalid and needs to be fixed.
```
--- a/airflow/providers/qubole/operators/qubole_check.py
+++ b/airflow/providers/qubole/operators/qubole_check.py
@@ -101,7 +101,7 @@ class QuboleCheckOperator(CheckOperator, QuboleOperator):
         if hasattr(self, 'hook') and (self.hook is not None):
             return self.hook
         else:
-            return QuboleCheckHook(context=context, *self.args, **self.kwargs)
+            return QuboleCheckHook(context=context, **self.kwargs)

     def __getattribute__(self, name):
         if name in QuboleCheckOperator.template_fields:
```

**Review by mik-laj — APPROVED** [review]

**Review by turbaszek — APPROVED** [review]

**Pull request merged**